File: src/main/java/cn/java/ConstructorOrder.java
package cn.java;

public class ConstructorOrder {

    {
        System.out.println("blank code");
    }

    static {
        System.out.println("static blank code");
    }

    public ConstructorOrder() {
        System.out.println("constructor");
    }

    public static void main(String[] args) {
        new ConstructorOrder();
    }
}

File: src/main/java/cn/kotlin/ConstructorOrder.kt
package cn.kotlin

class ConstructorOrder
constructor(name: String) {
    constructor() : this("asdf") {
        println("second constructor")
    }
    init {
        println("init $name")
    }
    init {
        println("init 2")
    }
    init {
        println("init 3")
    }
}

fun main() {
    ConstructorOrder()
}